src/decode.rs

```diff
@@ -4954,7 +4966,11 @@ pub(crate) unsafe fn rav1d_decode_frame(c: &Rav1dContext, fc: &Rav1dFrameContext
         }
     }
     rav1d_decode_frame_exit(c, fc, res);
-    res
+    if let Some(err) = *fc.task_thread.retval.try_lock().unwrap() {
+        Err(err)
+    } else {
+        Ok(())
+    }
```

Comment:
```suggestion
    fc.task_thread.retval.try_lock().unwrap().err_or(())
```

I added an extension method `.err_or` in `src/extensions.rs` for this.  It's annoying `std` doesn't have this.

Reply:
Also, just because we have a lock around `retval` now, can we return `retval` from `fn rav1d_decode_frame_exit`, too, and then just return that value here?

Reply:
we certainly can